src/nvim/option.c

```diff
@@ -2759,6 +2761,17 @@ did_set_string_option(
     if (check_ei() == FAIL) {
       errmsg = e_invarg;
     }
+  } else if (gvarp == &p_fcnotify) {  // 'filechangenotify'
+    if (check_opt_strings(curbuf->b_p_fcnotify, p_fcn_values, true) != OK) {
+      errmsg = e_invarg;
+    } else {
+      // TODO(BK1603): Why do tests don't respect local buffer?
+      if (strstr((char *)curbuf->b_p_fcnotify, "watcher") != NULL) {
+        do_cmdline((char_u *)"call v:lua.vim.fcnotify.init_watcher()", NULL, NULL, 0);
```

Comment:
You can execute this directly in Lua if you want.

This should also be "private" (leading underscore) if they should only be called fro core.

Reply:
Won't be necessary now since poll handle for watcher was removed. I will keep this in mind for when we add it later. :)